routes/room.go

```diff
@@ -96,40 +115,109 @@ func UpdateRoom(c *gin.Context) {
+		log.Printf("Failed to extract user from context: %v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to extract user from context"})
+		return
+	}
+
+	roomID := c.Param("id")
+
+	err = db.First(&room, roomID).Error
+	if err != nil {
+		log.Printf("Failed to find document: %v", err)
+		c.JSON(http.StatusNotFound, gin.H{"error": "Failed to find document"})
+		return
 	}
 
-	err = db.WithContext(ctx).Model(&patchedRoom).Updates(roomUpdate.ToRoom()).Error
+	roomUpdate.UsersID = append(roomUpdate.UsersID, user.ID)
+	err = db.WithContext(ctx).Model(&room).Updates(roomUpdate.ToRoom()).Error
 	if err != nil {
-		c.JSON(http.StatusInternalServerError, gin.H{"error": "Room not modified"})
 		log.Printf("Failed to modify document: %v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to modify document"})
 		return
 	}
 
-	c.JSON(http.StatusNoContent, nil)
+	c.JSON(http.StatusOK, room)
 }
 
-// DeleteRoom deletes a room in the database
-func DeleteRoom(c *gin.Context) {
+// DisconnectUserFromRoom disconnects a user from a room in the database
+func DisconnectUserFromRoom(c *gin.Context) {
+	var user *models.User
+	var room models.Room
+	var roomUpdate models.RoomUpdate
+
 	ctx, cancelCtx := context.WithTimeout(c, 1000*time.Millisecond)
 	defer cancelCtx()
 
-	id, err := strconv.Atoi(c.Param("id"))
+	user, err := utilsRoom.ExtractUserFromContext(c)
 	if err != nil {
-		c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid room ID"})
-		log.Printf("Failed to convert room ID: %v", err)
+		log.Printf("Failed to extract user from context: %v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to extract user from context"})
+		return
 	}
 
-	result := db.WithContext(ctx).Delete(&models.Room{}, id)
+	roomID := c.Param("id")
 
-	if result.Error != nil {
-		c.JSON(http.StatusInternalServerError, gin.H{"error": "Room not deleted"})
-		log.Printf("Failed to delete document: %v", result.Error)
-		return
-	} else if result.RowsAffected < 1 {
-		c.JSON(http.StatusNotFound, gin.H{"error": "Room not found"})
-		log.Printf("Failed to find document: %v", result.Error)
+	err = db.First(&room, roomID).Error
+	if err != nil {
+		log.Printf("Failed to find document: %v", err)
+		c.JSON(http.StatusNotFound, gin.H{"error": "Failed to find document"})
 		return
 	}
 
-	c.JSON(http.StatusNoContent, nil)
+	roomUpdate.UsersID = utils.RemoveUintFromSlice(roomUpdate.UsersID, user.ID)
+
+	// Check if user is owner to know if we give ownership to a new user
+	if room.OwnerID == user.ID {
+		if len(roomUpdate.UsersID) > 0 {
+			// We already removed the user from the slice, so we can get the first element as new Owner
+			roomUpdate.OwnerID = roomUpdate.UsersID[0]
+			err = db.WithContext(ctx).Model(&room).Updates(roomUpdate.ToRoom()).Error
+			if err != nil {
+				log.Printf("Failed to modify document: %v", err)
+				c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to modify document"})
+				return
+			}
```

Comment:
Externalize this behavior in another function. Will be useful to pass ownership of the room through another route later.